components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -603,4 +647,114 @@ - (void)didTapItemView:(UITapGestureRecognizer *)tap {
+  UITabBarItem *item = self.items[[self.itemViews indexOfObject:itemView]];
+  MDCTabBarViewPrivateIndicatorContext *context =
+      [[MDCTabBarViewPrivateIndicatorContext alloc] initWithItem:item
+                                                          bounds:selectionIndicatorBounds
+                                                    contentFrame:contentFrame];
+
+  // Ask the template for attributes.
+  id<MDCTabBarViewIndicatorTemplate> template = self.selectionIndicatorTemplate;
+  MDCTabBarViewIndicatorAttributes *indicatorAttributes =
+      [template indicatorAttributesForContext:context];
+
+  // Update the selection indicator.
+  [self.selectionIndicatorView applySelectionIndicatorAttributes:indicatorAttributes];
+}
+
+/// Sets _selectionIndicator's bounds and center to display under the item at the given index with
+/// no animation. May be called from an animation block to animate the transition.
+- (void)updateSelectionIndicatorToIndex:(NSUInteger)index {
+  if (index == NSNotFound || index >= self.items.count) {
+    // Hide selection indicator.
+    self.selectionIndicatorView.bounds = CGRectZero;
+    return;
+  }
+
+  // Place selection indicator under the item's cell.
+  CGRect selectedItemFrame = [self selectedItemView].frame;
+  if (CGRectEqualToRect(selectedItemFrame, CGRectZero)) {
+    selectedItemFrame =
+        [self estimatedFrameForItemAtIndex:[self.items indexOfObject:self.selectedItem]];
+  }
+  self.selectionIndicatorView.frame = selectedItemFrame;
+
+  CGRect selectionIndicatorBounds =
+      CGRectMake(0, 0, CGRectGetWidth(self.selectionIndicatorView.bounds),
+                 CGRectGetHeight(self.selectionIndicatorView.bounds));
+
+  // Extract content frame from item view.
+  CGRect contentFrame = selectionIndicatorBounds;
+  UIView *itemView = self.itemViews[index];
+  if ([itemView conformsToProtocol:@protocol(MDCTabBarViewIndicatorSupporting)]) {
+    UIView<MDCTabBarViewIndicatorSupporting> *supportingView =
+        (UIView<MDCTabBarViewIndicatorSupporting> *)itemView;
+    contentFrame = supportingView.contentFrame;
+  }
+
+  // Construct a context object describing the selected tab.
+  UITabBarItem *item = self.items[index];
+  MDCTabBarViewPrivateIndicatorContext *context =
+      [[MDCTabBarViewPrivateIndicatorContext alloc] initWithItem:item
+                                                          bounds:selectionIndicatorBounds
+                                                    contentFrame:contentFrame];
+
+  // Ask the template for attributes.
+  id<MDCTabBarViewIndicatorTemplate> template = self.selectionIndicatorTemplate;
+  MDCTabBarViewIndicatorAttributes *indicatorAttributes =
+      [template indicatorAttributesForContext:context];
+
+  // Update the selection indicator.
+  [self.selectionIndicatorView applySelectionIndicatorAttributes:indicatorAttributes];
+}
+
+- (void)didSelectItemAtIndex:(NSUInteger)index animateTransition:(BOOL)animate {
```

Comment:
Nit - Can you add a comment explaining this method? Specifically what happens when the index is NotFound...

Reply:
Done.